tests/Unit/DataStructures/Tensor/Expressions/Test_Divide.cpp

```diff
@@ -0,0 +1,293 @@
+void test_divide_rank0_denominator(const gsl::not_null<Generator*> generator,
+                                   const DataType& used_for_size) noexcept {
+  std::uniform_real_distribution<> distribution(0.1, 2.0);
+  constexpr size_t dim = 3;
+
+  const auto R = make_with_random_values<Scalar<DataType>>(
+      generator, distribution, used_for_size);
+  const auto S = make_with_random_values<tnsr::Ij<DataType, dim>>(
+      generator, distribution, used_for_size);
+  const auto T = make_with_random_values<tnsr::iJ<DataType, dim>>(
+      generator, distribution, used_for_size);
+
+  // \f$L_{i}{}^{j} = S^{i}{}_{j} / R\f$
+  // Use explicit type (vs auto) for LHS Tensor so the compiler checks the
+  // return type of `evaluate`
+  const tnsr::iJ<DataType, dim> result1 =
+      TensorExpressions::evaluate<ti_j, ti_I>(S(ti_I, ti_j) / R());
+
+  for (size_t i = 0; i < dim; i++) {
+    for (size_t j = 0; j < dim; j++) {
+      CHECK(result1.get(j, i) == S.get(i, j) / get(R));
+    }
+  }
+
+  // \f$L^{k}{}_{i} = (R T_{i}{}^{k}) / (T_{j}{}^{l} S^{j}{}_{l})\f$
+  const tnsr::Ij<DataType, dim> result2 =
+      TensorExpressions::evaluate<ti_K, ti_i>(
+          (R() * T(ti_i, ti_K)) / ((T(ti_j, ti_L) * S(ti_J, ti_l))));
+
+  DataType result2_expected_denominator =
+      make_with_value<DataType>(used_for_size, 0.0);
+  for (size_t j = 0; j < dim; j++) {
+    for (size_t l = 0; l < dim; l++) {
+      result2_expected_denominator += T.get(j, l) * S.get(j, l);
+    }
+  }
+
+  for (size_t i = 0; i < dim; i++) {
+    for (size_t k = 0; k < dim; k++) {
+      CHECK_ITERABLE_APPROX(
+          result2.get(k, i),
+          get(R) * T.get(i, k) / result2_expected_denominator);
+    }
+  }
+
+  // \f$L_{i}{}^{k} = T_{i}{}^{k} / R^2 / R\f$
+  const tnsr::iJ<DataType, dim> result3 =
+      TensorExpressions::evaluate<ti_i, ti_K>(T(ti_i, ti_K) / square(R()) /
+                                              R());
+
+  DataType result3_expected_denominator = get(R) * get(R) * get(R);
+  for (size_t i = 0; i < dim; i++) {
+    for (size_t k = 0; k < dim; k++) {
+      CHECK_ITERABLE_APPROX(result3.get(i, k),
+                            T.get(i, k) / result3_expected_denominator);
+    }
+  }
+}
+
+// \brief Test the division of a tensor expression over a rank 0 tensor
+// expression is correctly evaluated when generic indices are used for some
+// spacetime indices
+//
+// \details
+// The cases tested are:
+// - \f$L_{ai} = (T_{a}{}^{i} - S^{i}{}_{a}) / R\f$
+// - \f$L = (R / (T_{j}{}^{l} S^{j}{}_{l}) / 2\f$
+//
+// where \f$R\f$, \f$S\f$, \f$T\f$, and \f$L\f$ are Tensors with data type
+// `double` or DataVector.
+//
+// \tparam DataType the type of data being stored in the operands and result
+template <typename Generator, typename DataType>
+void test_divide_spatial_spacetime_index(
+    const gsl::not_null<Generator*> generator,
+    const DataType& used_for_size) noexcept {
+  std::uniform_real_distribution<> distribution(-2.0, 2.0);
```

Comment:
This is also used in the denominator (line 206 below)

Reply:
Fixed in new fixup!